addons/hr_expense/models/hr_expense.py

```diff
@@ -156,6 +156,42 @@ def _onchange_product_uom_id(self):
         if self.product_id and self.product_uom_id.category_id != self.product_id.uom_id.category_id:
             raise UserError(_('Selected Unit of Measure does not belong to the same category as the product Unit of Measure.'))
 
+    def create_expense_from_attachments(self, attachment_ids=[]):
```

Comment:
1/ Did you consider the security ? You have a public method to create attachments from an arbitrary list of ids.
2/ You should never use a mutable object as a method parameter. See: https://www.tutorialdocs.com/article/9-worst-python-practices.html

Reply:
@tivisse 
For point 1: I have put a condition, in that I have checked res_id and res_model values. If condition will be true then only 'create' will execute and records will be created.
For point 2: changes are done !